subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -0,0 +1,1212 @@
+                let icons = ureq::get("https://assets.blockstream.info/icons.json").call();
+                if registry.status() == 200 && icons.status() == 200 {
+                    match (registry.into_json(), icons.into_json()) {
+                        (Ok(mut registry), Ok(icons)) => {
+                            info!("got registry and icons");
+                            if let Some(policy) = registry_policy {
+                                info!("inserting policy asset {}", &policy);
+                                registry[policy] = json!({"asset_id": &policy, "name": "Liquid Bitcoin", "ticker": "L-BTC"});
+                            }
+
+                            db_for_registry.insert_asset_registry(&registry).unwrap();
+                            db_for_registry.insert_asset_icons(&icons).unwrap();
+                        }
+                        _ => warn!("Registry or icons are not json"),
+                    }
+                } else {
+                    warn!("Cannot download registry and icons");
+                }
+            }))
+        } else {
+            None
+        };
+
+        let checker = match self.network.id() {
+            NetworkId::Bitcoin(network) => {
+                let mut path: PathBuf = self.data_root.as_str().into();
+                path.push(format!("headers_chain_{}", network));
+                ChainOrVerifier::Chain(HeadersChain::new(path, network)?)
+            }
+            NetworkId::Elements(network) => {
+                let verifier = Verifier::new(network);
+                ChainOrVerifier::Verifier(verifier)
+            }
+        };
+
+        let mut headers_kind = match &self.url {
+            ElectrumUrl::Tls(url, validate) => {
+                let client = electrum_client::Client::new_ssl(url.as_str(), *validate)?;
+                HeadersKind::Tls(
+                    Headers::new(db.clone(), checker, client)?,
+                    url.to_string(),
+                    *validate,
+                )
+            }
+            ElectrumUrl::Plaintext(url) => {
+                let client = electrum_client::Client::new(&url)?;
+                HeadersKind::Plain(Headers::new(db.clone(), checker, client)?, url.to_string())
+            }
+        };
+        let (close_headers, r) = channel();
+        self.closer.senders.push(close_headers);
+        let mut chunk_size = DIFFCHANGE_INTERVAL as usize;
+        let headers_handle = thread::spawn(move || {
+            info!("starting headers thread");
+            'outer: loop {
+                if wait_or_close(&r, sync_interval) {
+                    info!("closing headers thread");
+                    break;
+                }
+
+                loop {
+                    match headers_kind.ask(chunk_size) {
+                        Ok(headers_found) => {
+                            if headers_found == 0 {
+                                chunk_size = 1
```

Comment:
Once the bug in esplora (mentioned in the comment below) is fixed, maybe better to increase this from 1? So if the client was idle for awhile, it'll get the new blocks in one go and won't have to wait for the `sync_interval` (7 seconds) to catch up with each one.

Reply:
When the bug is fixed, I think we can leave the chunk_size always 2016?

Reply:
Yeah, I think so to. I'll try to get it fixed soon.

Reply:
Fixed in https://github.com/Blockstream/electrs/commit/ff9237df4223468d4f1260cc377a013cf3a1ed25.